### PR TITLE
Bump Node.js to 18.x, start testing with 20.x

### LINKED ISSUE #12529: Switch to Node 18.x by default
Part of https://github.com/mozilla/addons/issues/1536

┆Issue is synchronized with this [Jira Task](https://mozilla-hub.atlassian.net/browse/ADDFRNT-257)

### PR DESCRIPTION
**Pull request by diox**
Fixes #12529

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -203,7 +203,7 @@ jobs:
     <<: *defaults-release
     steps:
       - better_checkout
-      - run: pyenv global 3.10.2
```
Comment (eviljeff) [inline .circleci/config.yml:206]: Is this change intended?  We're still on 3.10 in addons-server.
(and so on for the other bumps in this file.)

**Review by eviljeff — APPROVED** [review]

**Review by willdurand — APPROVED** [review]

File: .circleci/config.yml (hunk 1 above)
Comment (diox) [reply .circleci/config.yml:206]: Yes, it's because I've updated the ubuntu image to keep up with security updates but Python 3.10 is not available that updated image. This is just to run `dennis` so the mismatch doesn't matter much.
